assets/js/src/forms/list.jsx

```diff
@@ -127,25 +127,12 @@ const itemActions = [
 
 class FormList extends React.Component {
   createForm = () => {
-    MailPoet.Ajax.post({
-      api_version: window.mailpoet_api_version,
-      endpoint: 'forms',
-      action: 'create',
-    }).done((response) => {
-      MailPoet.trackEvent('Forms > Add New', {
-        'MailPoet Free version': window.mailpoet_version,
-      });
-      setTimeout(() => {
-        window.location = window.mailpoet_form_edit_url + response.data.id;
-      }, 200);
-    }).fail((response) => {
-      if (response.errors.length > 0) {
-        MailPoet.Notice.error(
-          response.errors.map((error) => error.message),
-          { scroll: true }
-        );
-      }
+    MailPoet.trackEvent('Forms > Add New', {
+      'MailPoet Free version': window.mailpoet_version,
     });
+    setTimeout(() => {
+      window.location = window.mailpoet_form_edit_url;
+    }, 200); // leave some time for the event to track
```

Comment:
I see the timeout is not a new addition but why don't we just wait for the promise? And is it needed (since we don't care how the request ends)?

Reply:
Because there is no promise there in the `MailPoet.trackEvent`. We are using mixpanel library and it doesn't return a promise: https://developer.mixpanel.com/docs/javascript-full-api-reference#mixpaneltrack so I don't see a better way here.